CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+```
+
+<!-- ### Release Procedure (For Package Maintainers)
+
+Follow gitflow best practices as linked above.
+
+1. Always merge all new features into `develop` branch first (unless it is a documentation, readme, or github action patch into `main`)
+1. After initial features are ready in the `develop` branch, create a `release-X.X` branch to prepare for the release.
+    1. update the `__version__` parameter in `genie/__init__.py`
+1. Merge `release-X.X` branch into `main` - Not by pull request!
+1. Create release tag (`v...`) and include release notes.  Also include any known bugs for each release here.
+1. Merge `main` back into `develop` -->
+
+### Transforms
+
+This package has a `src/agoradatatools/etl/transform` submodule.  This folder houses all the individual transform modules required for the package.  Here are the steps to add more transforms:
+
+1. Create new script in the transform submodule that matches the dataset name and name the function `transform_...`.  For example, if you have a dataset named `genome_variants`, your new script would be `src/agoradatatools/etl/transform/transform_genome_variants.py`.
+1. Register the new transform function in `src/agoradatatools/etl/transform/__init__.py`. Look in that file for examples.
+1. Modify the `apply_custom_transformations` in `src/agoradatatools/process.py` to include your new transform.
+1. Write a test for the transform:
+    - For transform tests, we are using a [Data-Driven Testing](https://www.develer.com/en/blog/data-driven-testing-with-python/) strategy
+    - To contribute new tests, assets in the form of input and output data files are needed.
+        - The input file is loaded to serve as the data fed into the transform function, while the output file is loaded in to check the function output against.
+    - These tests should include multiple ways of evaluating the transform function, including one test that should pass (good input data) and at least one that should fail (bad input data).
+    - For some functions, it may be appropriate to include multiple passing datasets (e.g. for functions that are written to handle imperfect data) and/or multiple failing datasets (e.g. for transforms operating on datasets that can be unclean in multiple distinct ways).
+    - Each transform function should have its own folder in `test_assets` to hold its input and output data files. Inputs should be in CSV form and outputs in JSON form.
+    - Use `pytest.mark.parameterize` to loop through multiple datasets in a single test.
+    - The class `TestTransformGenesBiodomains` can be used as an example for future tests contibuted.
+
+### DockerHub
+
+This repository does not use github actions to push docker images.  By adding the `sagebiodockerhub` github user as an Admin to this GitHub repository, we can configure an automated build in DockerHub.  You can view the repo [here](https://hub.docker.com/r/sagebionetworks/agora-data-tools).
```

Comment:
This confused me a bit, but I think it's saying:

Rather than using github actions to build and push docker images to DockerHub, the docker images are automatically built on DockerHub. This requires that the `sagebiodockerhub` user is an Admin of this repo.

Did I interpret that correctly?

Reply:
Yes, that is the correct interpretation. But I'd actually like us to switch to use gh-actions to build to gh-packages so that everything in contained in the repo.